### DISCUSSION TITLE
Use ignoreRestSiblings on no-unused-vars in eslint:recommended rules

### DISCUSSION THREAD

**Discussion by IchordeDionysos** [eslint/eslint#18488] [Ideas]
Not sure where the best place to ask is or what the process is to suggest changes to the recommended rules.

But I believe that enabling `ignoreRestSiblings` on `no-unused-vars` can make people's lives easier :)

**Comment by Rec0iL99** [comment]
Hi @IchordeDionysos, you can override the recommended rules setting in your config. I think the main focus of the recommended rules config is for providing a basic starting point for users to extend from.

```js
// eslint.config.js
{
    rules: {
      "no-unused-vars": ["error", { ignoreRestSiblings: true }],
    },
  },
```

**Reply by IchordeDionysos** [reply]
@Rec0iL99 I understand that it should be a basic starting point.

From multiple usages that I see is that the rest pattern is commonly used, see here:
https://javascript.info/destructuring-assignment#the-rest-pattern

A code like this:

```ts
let options = {
  title: "Menu",
  height: 200,
  width: 100
};

// title = property named title
// rest = object with the rest of properties
let {title, ...rest} = options;

// now title="Menu", rest={height: 200, width: 100}
alert(rest.height);  // 200
alert(rest.width);   // 100
```

Where you want to discard title is something that restructuring really shines. Especially in Typescript environments, where doing something like `delete options.title` might not be as easy to do with strict typing.

**Reply by IchordeDionysos** [reply]
With the current "recommended" settings the rest pattern is basically discouraged

**Reply by Rec0iL99** [reply]
I kinda disagree with that. The recommended rules config is just turning on some recommended core rules as a basic starting point that users can extend from. 

https://eslint.org/docs/latest/use/configure/configuration-files#using-predefined-configurations
https://github.com/eslint/eslint/blob/f6534d14033e04f6c7c88a1f0c44a8077148ec6b/packages/js/src/configs/eslint-recommended.js#L74
